ios/PusherWebsocketReactNative.swift

```diff
@@ -99,17 +102,29 @@ import Foundation
         ])
         
         let key = channelName + socketID
-        authorizerMutex[key] = DispatchSemaphore(value: 0)
-        authorizerMutex[key]!.wait()
-        let authParams = authorizerResult.removeValue(forKey: key)!
-        completionHandler(PusherAuth(auth: authParams["auth"]!, channelData: authParams["channel_data"], sharedSecret: authParams["shared_secret"]))
```

Comment:
Here we assumed that `authParams["auth"]!` is always defined, while nothing prevents the engineer to return an empty object from JS, so I added an explicit check.

We could pass `nil` instead of this `PusherAuth` with `auth: "<missing_auth_param>:error"`, but this is the easiest way to trigger a readable error in the current architecture. Passing `nil` would simply drop the auth and the subscription without any error.

I'm open to suggestions how to improve this, but I focused on removing the mutex in this PR, not error handling